scripts/gen-did/main.ts

```diff
@@ -89,88 +78,91 @@ function exportKeypairs(keypairs: {
+    console.debug(`Connected to ${endpoint}`)
+    // console.log('api', api.on)
     // get first command line argument as seed for the payment account and fail if not provided
-    const paymentSeed = process.argv[2];
+    const paymentSeed = process.argv[2]
+    console.debug(paymentSeed)
     if (!paymentSeed) {
         // console.error('Please provide a seed for the payment account as first command line argument.');
-        process.exit(1);
+        console.error('Payment seed not found')
+        process.exit(1)
     }
-    let paymentKeyPair = Kilt.Utils.Crypto.makeKeypairFromSeed(
-        mnemonicToMiniSecret(paymentSeed),
-        'sr25519',
-    );
-    console.debug(`Payment account address: ${paymentKeyPair.address}`);
+    const didSeed = process.argv[2]
+
+    const paymentKeyPair = Kilt.Utils.Crypto.makeKeypairFromUri(paymentSeed, 'sr25519')
+    console.debug(`Payment account address: ${paymentKeyPair.address}`)
     // get the balance of the payment account
-    const accountInfo = await api.query.system.account(paymentKeyPair.address);
-    console.debug(`Payment account balance: ${accountInfo.data.free}`);
-
-    const keypairs = generateKeypairs();
-    const didSecrets = exportKeypairs(keypairs);
-    await writeDidSecretsFile(didSecrets, 'did-secrets.json');
-    console.debug(`Wrote did-secrets.json`);
-
-    let signCallback = (signData: Kilt.SignRequestData): Promise<Kilt.SignResponseData> => {
-        return new Promise<Kilt.SignResponseData>((resolve, reject) => {
-            const signResponseData: Kilt.SignResponseData = {
-                signature: keypairs.authentication.sign(signData.data),
-                keyType: keypairs.authentication.type,
-                keyUri: `did:kilt:${keypairs.authentication.address}#lolidontknow`,
-            };
-            resolve(signResponseData);
-        });
-    };
-
-    // register the DID on chain
-    console.debug(`Registering DID ${keypairs.authentication.address} on chain`);
-    const fullDidCreationTx = await Kilt.Did.getStoreTx({
-        authentication: [{publicKey: keypairs.authentication.publicKey, type: 'sr25519'}],
-        assertionMethod: [{publicKey: keypairs.assertionMethod.publicKey, type: 'sr25519'}],
-        keyAgreement: [{publicKey: keypairs.keyAgreement.publicKey, type: 'x25519'}],
-    },
-        paymentKeyPair.address,
-        signCallback
-    );
-    try {
-        await Kilt.Blockchain.signAndSubmitTx(fullDidCreationTx, paymentKeyPair);
-        console.debug(`Registered DID ${keypairs.authentication.address} on chain`);
-    } catch (error) {
-        // console.error(error);
-        process.exit(1);
+    const accountInfo = await api.query.system.account(paymentKeyPair.address)
+    console.debug(`Payment account balance: ${accountInfo.data.free}`)
+
+    const keypairs = generateKeypairs(didSeed)
+    const didSecrets = exportKeypairs(keypairs)
+    await writeDidSecretsFile(didSecrets, 'did-secrets.json')
```

Comment:
This is dangerous since it will overwrite already existing mnemonics. Possibly resulting in the loss of the mnemonic.

Reply:
Agreed. I have made this into a new function and updated the logic.